lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +36,59 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        #meta_files = list(self.path_to_raw_txt_data.glob('*_meta.json'))
+        raw_files = list(self.path_to_raw_txt_data.glob('*_raw.txt'))
+        #meta_files_list = [str(x) for x in meta_files]
+        raw_files_list = [str(x) for x in raw_files]
+        max_number = len(raw_files_list)
+        path = str(self.path_to_raw_txt_data)
+
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if max_number == 0:
+            raise EmptyDirectoryError
+
+        #if len(meta_files_list) != len(raw_files_list):
+        #    raise InconsistentDatasetError
+        #for file in meta_files:
+        #    if not file.stat().st_size:
+        #        raise InconsistentDatasetError
+        for file in raw_files:
+            if not file.stat().st_size:
+                raise InconsistentDatasetError
+        for i in range(max_number):
+            filename_raw = path + '\\' + str(i+1) + '_raw.txt'
+        #    filename_meta = path + '\\' + str(i + 1) + '_meta.json'
+            if filename_raw not in raw_files_list: #or filename_meta not in meta_files_list:
```

Comment:
you may omit comments with not used code

Reply:
just to shorten implementation

Reply:
I want to use this code later, as there are more checks in this method for higher marks